src/django_upgrade/fixers/admin_register.py

```diff
@@ -130,56 +133,103 @@ def visit_Call(
+        ):
+            if isinstance(first_arg, ast.Name):
+                model_names = {first_arg.id}
+            else:
+                # cast() could be removed by using TypeGuard func above
+                model_names = {cast(ast.Name, elt).id for elt in first_arg.elts}
+            admin_name = admin_arg.id
+            admin_details = decorable_admins.get(state, {}).get(admin_name, None)
+            unregistered_models = unregistered_site_models.get(state, {}).get(
+                site_name, set()
             )
-        )
-        and (
-            isinstance((first_arg := node.args[0]), ast.Name)
-            or (
-                isinstance(first_arg, (ast.Tuple, ast.List))
-                and all(isinstance(elt, ast.Name) for elt in first_arg.elts)
+            num_intersections = len(model_names & unregistered_models)
+            if (
+                # We do not rewrite register calls that contain at least one
+                # unregistered model.
+                num_intersections == 0
+                and admin_details is not None
+                and admin_details.parent == parents[-2]
+                and not (site_name and not admin_name.endswith("Admin"))
+            ):
+                admin_details.model_names_per_site.setdefault(site_name, set()).update(
+                    model_names
+                )
+                yield ast_start_offset(node), partial(erase_node, node=parents[-1])
+        elif (
+            node.func.attr == "unregister"
+            and (
+                (  # admin.site.unregister(...)
+                    isinstance(node.func.value, ast.Attribute)
+                    and node.func.value.attr == "site"
+                    and isinstance(node.func.value.value, ast.Name)
+                    and node.func.value.value.id == "admin"
+                    and (site_name := "") == ""
+                )
+                or (  # custom_site.unregister(...)
+                    isinstance(node.func.value, ast.Name)
+                    and (site_name := node.func.value.id).endswith("site")
+                    and state.looks_like_admin_file()
+                )
             )
-        )
-    ):
-        if isinstance(first_arg, ast.Name):
-            model_names = {first_arg.id}
-        else:
-            # cast() could be removed by using TypeGuard func above
-            model_names = {cast(ast.Name, elt).id for elt in first_arg.elts}
-        admin_name = admin_arg.id
-        admin_details = decorable_admins.get(state, {}).get(admin_name, None)
-
-        if (
-            admin_details is not None
-            and admin_details.parent == parents[-2]
-            and not (site_name and not admin_name.endswith("Admin"))
-        ):
-            admin_details.model_names_per_site.setdefault(site_name, set()).update(
-                model_names
+            and (  # unregister(…) has only one parameter
+                isinstance((first_arg := node.args[0]), ast.Name)
```

Comment:
this assumes that the function is called with pos args, not kwargs

A line like:

```py
admin.site.unregister(model_or_iterable=MyModel1)
```

crashes this code, because there's no preceding check that `node.args` has a length of at least 1

I'm gonna update this code, whilst also doing the below update.

Reply:
Oh, thank you. I've completely forgot about keyword arguments in this case.